src/main/kotlin/app/Main.kt

```diff
@@ -3,60 +3,112 @@
 
 package app
 
+import app.ui.ConsoleUi
 import app.utils.CommandConfig
-import app.utils.CommandExplore
+import app.utils.CommandAdd
+import app.utils.CommandList
+import app.utils.CommandRemove
 import app.utils.Options
+import app.utils.RepoHelper
 import com.beust.jcommander.JCommander
-import com.beust.jcommander.ParameterException
-import kotlin.system.exitProcess
 
 fun main(argv: Array<String>) {
-    val commandExplore = "explore"
+    val commandAdd = "add"
     val commandConfig = "config"
+    val commandList = "list"
+    val commandRemove = "remove"
 
     val options = Options()
-    val explore = CommandExplore()
+    val add = CommandAdd()
     val config = CommandConfig()
+    val list = CommandList()
+    val remove = CommandRemove()
     val jc: JCommander = JCommander.newBuilder()
             .programName("sourcerer")  // Used for usage method.
             .addObject(options)
-            .addCommand(commandExplore, explore)
+            .addCommand(commandAdd, add)
             .addCommand(commandConfig, config)
+            .addCommand(commandList, list)
+            .addCommand(commandRemove, remove)
             .build()
 
-    Configurator.setCurrentOptions(options)
+    jc.parse(*argv)
+    Configurator.options = options
 
-    try {
-        jc.parse(*argv)
-    } catch (e: ParameterException) {
-        println(e.message)
-        exitProcess(1)  // Reporting failure with non-zero exit code.
+    if (options.help) {
+        help(jc)
+        return
+    }
+
+    if (options.setup) {
+        setup()
+        return
     }
 
     when (jc.parsedCommand) {
-        commandExplore -> explore(explore)
+        commandAdd -> add(add)
         commandConfig -> config(config)
-        else -> help(jc)  // Show help info if no command specified.
+        commandList -> list(list)
+        commandRemove -> remove(remove)
+        else -> ui()
     }
-
-    println(arrayOf(options.username, options.password).joinToString())
-    println(jc.parsedCommand)
-    println(explore.path)
 }
 
-fun explore(commandOptions: CommandExplore) {
-    RepoExplorer(commandOptions).explore()
+fun add(commandAdd: CommandAdd) {
```

Comment:
Please use more explicit names for these functions, because it is getting confusing. use something like executeAdd, executeConfig, or doAdd, doConfig, so that it's easier to understand when reading the code that uses them.